File: SupportService/Domain/Entities/ServiceRequest.cs
namespace Domain.Entities;

public class ServiceRequest
{
    public int Id { get; set; }

    public string Title { get; set; }

    public string Description { get; set; }

    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedDate { get; set; }

    public int GroupId { get; set; }

    #region Navigation Properties

    public int CreatedById { get; set; }

    public User CreatedBy { get; set; }

    public int? AppointedId { get; set; }

    public User Appointed { get; set; }

    public Status Status { get; set; }

    public List<Comment> Comments { get; set; }

    public Group? Group { get; set; }

    #endregion
}

File: SupportServiceUi/src/app/interfaces/ServiceRequest.ts
import { Group } from "./Group"
import { Status } from "./Status"
import { UserPreview } from "./User"

export interface ServiceRequestPreview {
    id: number,
    title: string,
    status: string
}

export interface ServiceRequestOverview {
    id: number,
    title: string,
    description: string,
    status: string,
    createdDate: Date,
    updatedDate: Date
    createdBy: string,
    appointed: string
}

export interface EditServiceRequest {
    id: number,
    title: string,
    description: string,
    status: Status,
    appointed: UserPreview | null,
    group: Group | null
}

export interface CreateServiceRequestDto {
    title: string,
    description: string,
    groupId: number | null,
    createdById: number,
    appointedId: number | null
}